Cesium.Runtime/StringFunctions.cs

```diff
@@ -186,19 +170,21 @@ public static int StrNCmp(byte* lhs, byte* rhs, nuint count)
     }
     public static byte* StrChr(byte* str, int ch)
     {
-        if (str == null)
+        if (str != null)
         {
-            return null;
-        }
+            int match;
+            nuint offset = 0;
+            byte c = (byte)ch;
 
-        while (*str != 0)
-        {
-            if (*str == ch)
+            // TODO: Copy IndexOfNullByte impl. from CoreLib in a distant future
+            var span = new Span<byte>(str, int.MaxValue);
```

Comment:
A no-op once compiled (span is optimized away), we have to do this sadly because `CoreLib` keeps all the nice methods for working with raw ptrs internal.